### ISSUE TITLE
[1.21.4] Bootstrap registered PaintingVariant(Art) throw NullPointerException on Art.getKey()

### ISSUE THREAD

**Issue by SoulxluoS** [PaperSpigot/Paper#11763]
### Expected behavior

Everything worked well on version 1.21.3.

### Observed/Actual behavior

The method & class where the throw happens. Commit with the changes - https://github.com/PaperMC/paper/commit/71a367ec8489c8a61a4c009f6a3c536438707158

```java
public class CraftArt implements Art, Handleable<PaintingVariant> {
      <...>
      @NotNull
      @Override
      public NamespacedKey getKey() {
          if (true) return java.util.Objects.requireNonNull(org.bukkit.Registry.ART.getKey(this), () -> this + " doesn't have a key"); // Paper
          return this.key;
      }
      <...>
}
```

### Steps/models to reproduce

Register PaintingVariant via Bootstrap LifecycleManager. Call Art.getKey().

### Paper version

- **1.21.4-15-master@bb76819** (2024-12-15T09:48:21Z)
- Previous version: 1.21.3-81-da71382 (MC: 1.21.3)

- **[HardFork](https://github.com/PaperMC/testing/releases/tag/hard-fork) also have this bug.**

**Comment by electronicboy** [comment]
the art registry is not exposed in the lifecycle management stuff?

**Comment by YoungSoulluoS** [comment]
*PAINTING_VARIANT

**Comment by Machine-Maker** [comment]
Yeah, I was able to replicate it. I see the issue.

**Comment by Machine-Maker** [comment]
Fixes in 46c6f497c7e5e3f6d7ee9e226837d44211737590 (not deployed yet anywhere I don't think)
